radix-transaction-scenarios/src/scenarios/all_scenarios.rs

```diff
@@ -1,54 +1,422 @@
+        FungibleResourceScenarioCreator::create,
+        NonFungibleResourceScenarioCreator::create,
+        AccountAuthorizedDepositorsScenarioCreator::create,
+        GlobalNOwnedScenarioCreator::create,
+        NonFungibleResourceWithRemoteTypeScenarioCreator::create,
+        KVStoreScenarioCreator::create,
+        MaxTransactionScenarioCreator::create,
+        RoyaltiesScenarioCreator::create,
+    ],
+    // The set of scenarios that should be run after the anemone protocol update.
+    ScenarioRequirements::ProtocolUpdateUpTo(ProtocolUpdate::Anemone) => [],
+    // The set of scenarios that should be run after the bottlenose protocol update.
+    ScenarioRequirements::ProtocolUpdateUpTo(ProtocolUpdate::Bottlenose) => [
+        AccountLockerScenarioCreator::create
+    ],
+}
+
+pub struct TransactionScenarioExecutor<D, W, E, F1, F2>
+where
+    D: SubstateDatabase + CommittableSubstateDatabase,
+    W: WasmEngine,
+    E: NativeVmExtension,
+    F1: FnMut(&ScenarioMetadata, &NextTransaction, &TransactionReceiptV1, &D),
+    F2: FnMut(&ScenarioMetadata),
+{
+    /* Environment */
+    /// The substate database that the scenario will be run against.
+    database: D,
+    /// The Scrypto VM to use in executing the scenarios.
+    scrypto_vm: ScryptoVm<W>,
+    /// The Native VM to use in executing the scenarios.
+    native_vm: NativeVm<E>,
+
+    /* Execution */
+    /// The scenarios that the executor should execute. They are not specified per scenario but per
+    /// requirement. Meaning that through this field a client may opt-in or opt-out of the scenarios
+    /// of an entire protocol update but not individual ones.
+    scenarios_to_execute: BTreeSet<ScenarioRequirements>,
+    /// Controls whether the bootstrap process should be performed or not.
+    bootstrap: bool,
+    /// The first nonce to use in the execution of the scenarios.
+    starting_nonce: u32,
+    /// How the executor should handle nonces and how it should get the next nonce.
+    nonce_handling: NonceHandling,
+    /// The network definition to use in the execution of the scenarios.
+    network_definition: NetworkDefinition,
+
+    /* Callbacks */
+    /// A callback that is called when a scenario transaction is executed.
+    on_transaction_execution: F1,
+    /// A callback that is called when a new scenario is started.
+    on_scenario_start: F2,
+}
+
+pub type DefaultTransactionScenarioExecutor<D> = TransactionScenarioExecutor<
+    D,
+    DefaultWasmEngine,
+    NoExtension,
+    fn(&ScenarioMetadata, &NextTransaction, &TransactionReceiptV1, &D),
+    fn(&ScenarioMetadata),
+>;
+
+impl<D, W, E, F1, F2> TransactionScenarioExecutor<D, W, E, F1, F2>
+where
+    D: SubstateDatabase + CommittableSubstateDatabase,
+    W: WasmEngine,
+    E: NativeVmExtension,
+    F1: FnMut(&ScenarioMetadata, &NextTransaction, &TransactionReceiptV1, &D),
+    F2: FnMut(&ScenarioMetadata),
+{
+    pub fn new(database: D) -> DefaultTransactionScenarioExecutor<D> {
```

Comment:
Most of the defaults are reasonable, but I'd rather we take an explicit `network_definition` here.

Reply:
Yup agreed, seems like `network_definition` is the only one that's out of place there and the default doesn't make sense to all clients. Will move it to be an explicit parameter passed to `new`.